src/Core/Framework/Util/IOStreamHelper.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopware\Core\Framework\Util;
+
+use Shopware\Core\Framework\Log\Package;
+
+#[Package('framework')]
+final class IOStreamHelper
+{
+    public static function writeError(string $message, ?\Throwable $error = null): void
```

Comment:
we could add a new method capture which accepts an callable. while that callable runs, we write any output to an static class property and that capture method returns then the string.

```php
$output = IOStreamHelper::capture(function() {
    $service->doSomething();
});

static::assertSame('foo', $output);
```